packages/web/hooks/use-swap.ts

```diff
@@ -316,20 +315,12 @@ export function useSwapAssets({
     switchAssets,
   } = useToFromDenoms(useQueryParams, initialFromDenom, initialToDenom);
 
-  // get selectable currencies for trading, including user balances if wallect connected
-  const [assetsQueryInput, setAssetsQueryInput] = useState<string>("");
-
   // generate debounced search from user inputs
-  const [debouncedSearchInput, setDebouncedSearchInput] =
-    useDebouncedState<string>("", 500);
-  useEffect(() => {
-    setDebouncedSearchInput(assetsQueryInput);
-  }, [setDebouncedSearchInput, assetsQueryInput]);
-
-  const inputSearch = useMemo(
-    () => (debouncedSearchInput ? { query: debouncedSearchInput } : undefined),
-    [debouncedSearchInput]
-  );
+  const {
+    searchInput: assetsQueryInput,
+    setSearchInput: setAssetsQueryInput,
+    queryInput,
+  } = useSearchQueryInput();
```

Comment:
Thoughts on using `useDebouncedState` directly here instead of creating a separate hook for the search query input? My concern is that the separate hook might be overly complex for what we need to achieve: a denounced query state